ast/format_test.go

```diff
@@ -283,27 +320,79 @@ highestAverage = (n, columns=["_value"], by, tables=<-) =>
 		},
 	}
 
-	for _, tc := range testCases {
-		tc := tc
-		t.Run(tc.name, func(t *testing.T) {
-			if reason, ok := skip[tc.name]; ok {
-				t.Skip(reason)
-			}
-
-			pkg := parser.ParseSource(tc.script)
-			if ast.Check(pkg) > 0 {
-				err := ast.GetError(pkg)
-				t.Fatal(errors.Wrapf(err, "source has bad syntax:\n%s", tc.script))
-			}
+	formatTestHelper(t, testCases)
+}
 
-			stringResult := ast.Format(pkg.Files[0])
+func TestFormat_Associativity(t *testing.T) {
+	testCases := []formatTestCase{
+		{
+			name:   "math no pars",
+			script: `a * b + c / d - e * f`,
+		},
+		{
+			name:   "math with pars",
+			script: `(a * b + c / d - e) * f`,
+		},
+		{
+			name:   "math with more pars",
+			script: `(a * (b + c) / d / e * (f + g) - h) * i * j / (k + l)`,
+		},
+		{
+			name:   "logic",
+			script: `a or b and c`,
+		},
+		{
+			name:   "logic with pars",
+			script: `(a or b) and c`,
+		},
+		{
+			name:   "logic with comparison",
+			script: `a == 0 or b != 1 and c > 2`,
+		},
+		{
+			name:   "logic with comparison with pars",
+			script: `(a == 0 or b != 1) and c > 2`,
+		},
+		{
+			name:   "logic and math",
+			script: `a * b + c * d != 0 or not e == 1 and f == g`,
+		},
+		{
+			name:   "logic and math with pars",
+			script: `(a * (b + c) * d != 0 or not e == 1) and f == g`,
+		},
+		{
+			name:   "unary",
+			script: `not b and c`,
+		},
+		{
+			name:   "unary with pars",
+			script: `not (b and c)`,
+		},
+		{
+			name:   "function call with pars",
+			script: `(a + b * c == 0)(foo: "bar")`,
+		},
+		{
+			name:   "member with pars",
+			script: `((a + b) * c)._value`,
```

Comment:
Should this be a test case? I guess this parses just fine, but I don't think it would type check. Are there any member expression examples that are valid Flux expressions? I can't really think of any.

Reply:
Answered below.